qemu/tests/sr_iov_lifecycle.py

```diff
@@ -0,0 +1,109 @@
+        msg = "Unsupport device type '%s'." % device_type
+        msg += " Please set device_type to 'vf' or 'pf'."
+        test.cancel(msg)
+
+    msg = "Try to boot up guest(s) with VF(s)."
+    error_context.context(msg, logging.info)
+    timeout = int(params.get("login_timeout", 30))
+
+    for vm_name in params["vms"].split(" "):
+        params["start_vm"] = "yes"
+        vm = env.get_vm(vm_name)
+        session = vm.wait_for_serial_login(
+            timeout=int(params.get("login_timeout", 720)))
+        rc, output = session.cmd_status_output(
+            "ip li| grep -i 'BROADCAST'|awk '{print $2}'| sed 's/://'")
+        if not rc:
+            iface_probed = output.splitlines()
+            logging.info("probed VF Interface(s) in guest: %s",
+                         iface_probed)
+
+        if params.get("vm_suspend", "no") == "yes":
+            error_context.base_context("Suspending the VM", logging.info)
+            vm.pause()
+            error_context.context("Verify the status of VM is"
+                                  "'paused'", logging.info)
+            vm.verify_status("paused")
+
+            error_context.context("Verify the session has"
+                                  " no response", logging.info)
+            if session.is_responsive():
+                msg = "Session is still responsive after suspend"
+                logging.error(msg)
+                test.fail(msg)
+            session.close()
+
+            time.sleep(20)
+            error_context.base_context("Resuming the"
+                                       "VM", logging.info)
+            vm.resume()
+            error_context.context("Verify the status of VM is"
+                                  "'running'", logging.info)
+            vm.verify_status("running")
+
+            error_context.context("Re-login the guest", logging.info)
+            session = vm.wait_for_serial_login(
+                timeout=int(params.get("login_timeout", 360)))
+
+            error_context.context("Verify Host and guest kernel "
+                                  "no error and call trace", logging.info)
+            vm.verify_kernel_crash()
+
+        if params.get("vm_reboot", "no") == "yes":
+            adapter_list_before = session.get_command_output("lspci -nn")
+            error_context.context("Rebooting Guest", logging.info)
+            logging.info("Adapter details before "
+                         "reboot are %s", adapter_list_before)
+            session.cmd('reboot & exit', timeout=10, ignore_all_errors=True)
+            session = vm.wait_for_serial_login(
+                timeout=int(params.get("login_timeout", 720)))
+            adapter_list_after = session.get_command_output("lspci -nn")
+            logging.info("Adapter details after reboot "
+                         "are %s ",  adapter_list_after)
+            if adapter_list_before != adapter_list_after:
+                test.fail("Mismatch in adapter list, after reboot.")
```

Comment:
This is right if the `adapter_list_before` have the passthrough VF, so let us check this initially and use it after reboot and suspend/resume.

Reply:
This validation need to be added for suspend/resume too.

Reply:
@balamuruhans modified as suggested, please review. Thanks!!